### ISSUE TITLE
Embed comparison into `compare_versions` macro

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#11914]
The [`compare_versions` macro](https://crystal-lang.org/api/1.3.2/Crystal/Macros.html#compare_versions(v1:StringLiteral,v2:StringLiteral):NumberLiteral-instance-method) compares two semver-ish strings (using `SemanticVersion`) and returns a number literal indicating the relation between the two versions.
It is typically used like this: 
```cr
{% if compare_versions(LIB_VERSION, "4.0.0") > 0 %}
  # do something that only works with version 4 or higher
{% end %}
```

The syntax is reasonably well to understand. I think it's pretty clear what the condition means.

But: It could be better! `compare_versions` could receive a `Call` which represents the comparison between both version values directly - without the indirection of comparing the result to `0`.
This makes it more concise:

```cr
{% if compare_versions(LIB_VERSION > "4.0.0") %}
  # do something that only works with version 4 or higher
{% end %}
```

And it should be pretty trivial to implement (and support alongside the current semantics).

This user-code implementation as a macro def shows how it's supposed to work:
```cr
macro compare_versions(call)
  {{ compare_versions(call.receiver, call.args[0]) }} {{ call.name }} 0
end

p! compare_versions("10.0.0" > "2.0.0") # => true
```
We just need to implement that for the macro method in the compiler.

It's a small enhancement and certainly not super important on its own. But it's easy to do. And such little convenience features help improve developer experience.

**Comment by asterite** [comment]
I guess it's fine, though it would be the first time a macro method receives a call instead of arguments. It reads a bit strange 🤷  And the amount of times this is needed maybe doesn't justify this small enhancement.

**Comment by beta-ziliani** [comment]
Yeah, it looks weird to me. If we want to improve the current situation, I'd say we should bite the bullet and expose (a subset of) SemanticVersion inside the macro language.

**Comment by straight-shoota** [comment]
> I'd say we should bite the bullet and expose (a subset of) SemanticVersion inside the macro language.

I don't think that would have much value. The only thing you need to do with version strings in macro expressions is comparing them. That's what `compare_versions` is for. I don't think we need anything else.
This proposal is just about a different way to *use* `compare_versions` that makes use of the power of macros to have code that reads a bit more natural.

**Comment by straight-shoota** [comment]
`compare_versions` is fine as it is. We don't need to change anything about it.
